apis/python/src/tiledb/vector_search/ivf_pq_index.py

```diff
@@ -177,6 +178,14 @@ def create(
         raise ValueError(
             f"Number of dimensions ({dimensions}) must be divisible by num_subspaces ({num_subspaces})."
         )
+    if distance_metric != vspy.DistanceMetric.L2:
+        raise ValueError(
+            f"Distance metric {distance_metric} is not supported in IVF_PQ"
+        )
+
+    group = tiledb.Group(uri, "w")
+    group.meta["distance_metric"] = int(distance_metric)
+    group.close()
```

Comment:
We should do this in C++ instead of in Python for the type-erased indexes (IVF PQ and Vamana).